teps/0114-custom-tasks-beta.md

```diff
@@ -0,0 +1,342 @@
+set to `"true"` or `enable-api-fields` needs to be set to `"alpha"`.
+
+`Custom Tasks` have been available since [v0.19.0][v0.19.0] of Tekton Pipelines released in December 2020.
+Many `Custom Tasks` have been created to extend Tekton Pipelines functionality, including but not limited to:
+- [Task Loops][task-loops]: Runs a `Task` in a loop with varying `Parameter` values. 
+- [Pipeline Loops][pipeline-loops]: Runs a `Pipeline` in a loop with varying `Parameter` values.
+- [Common Expression Language][cel]: Provides Common Expression Language support in Tekton Pipelines.
+- [Wait][wait]: Waits a given amount of time, specified by a `Parameter` named "duration", before succeeding.
+- [Approvals][approvals]: Pauses the execution of `PipelineRuns` and waits for manual approvals.
+- [Pipelines in Pipelines][pipelines-in-pipelines]: Defines and executes a `Pipeline` in a `Pipeline`. 
+- [Task Group][task-group]: Groups `Tasks` together as a `Task`.
+- [Pipeline in a Pod][pipeline-in-pod]: Runs `Pipeline` in a `Pod`.
+
+As discussed in [TEP-0096][tep-0096], `Custom Tasks` are critical for extensibility in Tekton Pipelines, as such
+promoting `Custom Tasks` to beta is a blocker for releasing v1 of Tekton Pipelines.
+
+## Proposal
+
+In this section, we scope the promotion of `Custom Tasks` to beta. The work in [required](#required) must be done,
+while the work in [optional](#optional) may be done.
+
+### Required
+
+The work scoped here are blockers for promotion.
+
+#### API Changes
+
+##### v1alpha1 to v1beta1 + Run to CustomRun
+
+Move `v1alpha1.Run` to `v1beta1.CustomRun`.
+
+The migration from `v1alpha1` to `v1beta1` provides an opportunity to rename `Runs` to `CustomRuns`. Reasons for the 
+name change as discussed in [tektoncd/pipelines#5005][5005] and [API WG meeting][wg-notes] include.:
+- `PipelineRuns` execute `Pipelines` and `TaskRuns` execute `Tasks`, so it is consistent for `CustomRuns` to execute
+  `CustomTasks`.
+- The current name `Runs` may suggest that it's an interface that `PipelineRuns` and `TaskRuns` implement.
+
+##### Embedded Specifications
+
+Simplify embedded specifications in `Runs`.
+
+Today, [embedded specs][runs-spec-docs] are in `Run.Spec.Spec`. We propose aligning the embedded specs with the rest of
+the API such that it's `Run.Spec.CustomSpec`.
+
+  ```yaml
+  # before
+  apiVersion: tekton.dev/v1alpha1
+  kind: Run
+  metadata:
+    name: run-with-spec
+  spec:
+    spec:
+      apiVersion: example.dev/v1alpha1
+      kind: Example
+      spec:
+        field1: value1
+        field2: value2
+  # after
+  apiVersion: tekton.dev/v1beta1
+  kind: CustomRun
+  metadata:
+    name: customrun-with-spec
+  spec:
+    customSpec:
+      apiVersion: example.dev/v1beta1
+      kind: Example
+      spec:
+        field1: value1
+        field2: value2
+  ```
+
+Today, [references][runs-spec-docs] are in `Run.Spec.Ref`. We propose aligning the embedded specs with the rest of the
+API such that it's `Run.Spec.CustomRef`.
```

Comment:
Run -> CustomRun

Reply:
aligning the embedded specs -> aligning the custom task reference 🤣

Reply:
haha will need some time to get used to customrun